create_cluster.py

```diff
@@ -16,6 +16,7 @@
 from getopt import getopt
 from jinja2 import Template
 from tempfile import TemporaryFile
+from functools import reduce
```

Comment:
This change is specific for Python3. `reduce` is not a built-in function anymore in python3. Should we change the shebang to explicitly run under python3 then?

Reply:
Good point. I didn't think it was necessary just yet. Should I?

Reply:
Cool! If it still works for py2 I am fine without enforcing python3 usage.